hooks/shotgun_multiple_publishes.py

```diff
@@ -0,0 +1,62 @@
+# Copyright (c) 2013 Shotgun Software Inc.
+#
+# CONFIDENTIAL AND PROPRIETARY
+#
+# This work is provided "AS IS" and subject to the Shotgun Pipeline Toolkit
+# Source Code License included in this distribution package. See LICENSE.
+# By accessing, using, copying or modifying this work you indicate your
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights
+# not expressly granted therein are reserved by Shotgun Software Inc.
+
+"""
+Hook executed when there are multiple published files in a Version.
+This allows to determine which PublishedFile to pick in the list.
+
+This default implementation cycles through the `viewer_extensions`
+config parameter, and returns the first published file matching
+one of the extensions.
+
+This allows to select PublishedFiles based on the order of
+`viewer_extensions`.
+
+If none of the PublishedFiles match any of the extensions,
+return the first one in the list.
+"""
+import sgtk
+from sgtk import TankError
+
+HookBaseClass = sgtk.get_hook_baseclass()
+
+
+class ChoosePublishedFile(HookBaseClass):
+    def execute(self, published_files, **kwargs):
+        """
+        Return the first PublishedFile matching
+        `viewer_extensions` configuration parameter.
+
+        If no PublishedFile matches the extensions,
```

Comment:
What is the use case for this?

Reply:
Isn't it better to launch the first published file than not launching at all?

Reply:
Not sure... I think that if you specify some extensions for a viewer, you just want that and nothing else. But it seems the logic to map an extension to an app is in two places: the settings, and the hook 🤪

Reply:
and the current version of the app does allow to open, let's say, a jpg without it being specified in `viewer_extensions`, so if I disallow it for the multiple published files case, I need to also disallow it in the regular case, which does not seem to be the philosophy here.
I still think it's better to return the first one.